gcm_filters/kernels.py

```diff
@@ -38,6 +39,36 @@ def _prepare_tripolar_exchanges(field):
 
 @dataclass
 class BaseScalarLaplacian(ABC):
+    """̵Base class for scalar Laplacians."""
+
+    def __call__(self, field):
+        pass  # pragma: no cover
+
+    # change to property when we are using python 3.9
+    # https://stackoverflow.com/questions/128573/using-property-on-classmethods
+    @classmethod
+    def required_grid_args(self):
+        try:
+            return list(self.__annotations__)
+        except AttributeError:
+            return []
+
+
+@dataclass
+class BaseScalarLaplacianWithArea(ABC):
+    """̵Base class for scalar Laplacians that are for simple fixed factor filtering. Simple
+    fixed factor filtering operates on a uniform Cartesian grid with dx=dy=1 that is being transformed
+    from the original locally orthogonal grid. In practice, the coordinate transformation is reflected
+    by a multiplication / division by the cell area of the original grid. Note that this occurs at the
+    filter level, rather than the Laplacian level.
+
+    Attributes
+    ----------
+    area: cell area
+    """
+
+    area: ArrayType
+
```

Comment:
There should be a way to make the class itself do the area weighting / deweighting. To me that would be a lot cleaner than doing it manually in the `filter` module (better separation of concerns).